### PR TITLE
Add IAM system tests

### PR DESCRIPTION
**Pull request by donatello**
## Description
- For internal IDP user, policy and groups

## Motivation and Context
- Improve test coverage and reduce technical debt.

## How to test this PR?
- No changes to behavior, tests should run in CI automatically.

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] enhancement

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Documentation updated
- [x] Unit tests added/updated

### CODE REVIEW

**Review by harshavardhana — APPROVED** [review]

File: internal/auth/credentials.go [hunk 1]
```diff
@@ -208,22 +209,28 @@ func GetNewCredentialsWithMetadata(m map[string]interface{}, tokenSecret string)
...
-	secretKey := strings.Replace(string([]byte(base64.StdEncoding.EncodeToString(keyBytes))[:secretKeyMaxLen]),
+	secretKey = strings.Replace(string([]byte(base64.StdEncoding.EncodeToString(keyBytes))[:secretKeyMaxLen]),
 		"/", "+", -1)
 
+	return accessKey, secretKey, nil
+}
+
+// GetNewCredentialsWithMetadata generates and returns new credential with expiry.
+func GetNewCredentialsWithMetadata(m map[string]interface{}, tokenSecret string) (cred Credentials, err error) {
+	accessKey, secretKey, err := GenerateCredentials()
```
Comment (vadmeste) [inline internal/auth/credentials.go:233]: You need to return an error here if err != nil

**Review by vadmeste — CHANGES_REQUESTED** [review]
One comment

File: internal/auth/credentials.go (hunk 1 above)
Comment (donatello) [reply internal/auth/credentials.go:233]: Good catch! Fixed.

**Review by kannappanr — APPROVED** [review]
LGTM

**Review by vadmeste — APPROVED** [review]
LGTM

**Pull request merged**